src/data/battlesService.js

```diff
@@ -4,8 +4,8 @@ import { processTeamModel } from './teamUtils'
 import { battle as runBattle } from 'gotchi-battler-game-logic'
 
 const processBattleModel = function (jsonData) {
-  const { team1, team2, ...restData } = jsonData
-  const teams = [team1 || null, team2 || null]
+  const { team1, team1Snapshot, team2, team2Snapshot, ...restData } = jsonData
+  const teams = [team1 || null, team2 || null, team1Snapshot || null, team2Snapshot || null]
```

Comment:
even simpler, if you see no need for the frontend to use the non-snapshot team, then you could just let the snapshot take priority here if present.  Or further up, in the server api endpoint.

Reply:
Yeah it felt a bit messy doing it this way but I was trying to take the approach of being as backwards compatible as possible by adding the snapshot data there to use if needed rather than replacing. It may well be the case the frontend never cares about the original team though. Off the top of your head can you think if the frontend ever retrieves the team information from the store by it's id rather than directly from an API call?